From 9712293bdafda3927b73e832d04ba8641946745e Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Sun, 24 Jan 2016 20:53:40 +0000
Subject: [PATCH] [CONFIGURATION-615] Adapted constructor of MapConfiguration.

Merged the change from CONFIGURATION-557 from version 1.10 to 2.0 branch.
This allows using MapConfiguration in the same way as in Configuration
1.7 and earlier.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/configuration/trunk@1726562 13f79535-47bb-0310-9956-ffa450edef68
---
 .../org/apache/commons/configuration2/MapConfiguration.java   | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/org/apache/commons/configuration2/MapConfiguration.java b/src/main/java/org/apache/commons/configuration2/MapConfiguration.java
index a1821f3bc5..12daed9d18 100644
--- a/src/main/java/org/apache/commons/configuration2/MapConfiguration.java
+++ b/src/main/java/org/apache/commons/configuration2/MapConfiguration.java
@@ -103,9 +103,9 @@ public class MapConfiguration extends AbstractConfiguration implements Cloneable
      *
      * @param map the map
      */
-    public MapConfiguration(Map<String, Object> map)
+    public MapConfiguration(Map<String, ?> map)
     {
-        this.map = map;
+        this.map = (Map<String, Object>) map;
     }
 
     /**
